src/Microsoft.Health.Dicom.Api/Features/Exceptions/ExceptionHandlingMiddleware.cs

```diff
@@ -25,6 +27,12 @@ namespace Microsoft.Health.Dicom.Api.Features.Exceptions;
 
 public class ExceptionHandlingMiddleware
 {
+    // SQL Errors taken from https://learn.microsoft.com/en-us/sql/relational-databases/errors-events/database-engine-events-and-errors-31000-to-41399?view=sql-server-ver16
+    private const int SQLKeyVaultCriticalError = 40981;
```

Comment:
As of now all the sql error codes are kept in health care shared components. Does it make sense to add here - https://github.com/microsoft/healthcare-shared-components/blob/main/src/Microsoft.Health.SqlServer/Features/Storage/SqlErrorCodes.cs

Reply:
Ah thanks I didn't know about this - will add it there. We are referencing these error codes in that repo anyways.

Reply:
there is already a static method `IsCMKError(this SqlException sqlEx)`